### DISCUSSION TITLE
Working with Lists

### DISCUSSION THREAD

**Discussion by PPetterssonSO** [SuperOffice/typescript-for-superoffice#27] [General]
Hi,
We are having trouble figuring out how to work with the list agents.

It always crashes, without hitting the catch.
For example:
try
{
var listAgent = soApi.getListAgent();
var listOfCountries = [752];
printLine("It passes this point"); // <-- custom print function..
const mmm = await listAgent.getCountryListAsync(listOfCountries); //Or any other function in listAgent..
printLine("It does not pass this point");

}
catch
{
  printLine("error"); // <-- custom print function..
}

Only returns: It passes this point
And, for more generic functions, like: getAllCountriesAsync

What is SO.List_GetAllCountriesRequests? that's required as a parameter?
How do we populate it, and with what?

SO.ListAgent.getAllCountriesAsync(request: SO.List_GetAllCountriesRequest, webapi_options?: base.WebApiRequestOptions): Promise<SO.Country[]> (+1 overload)

**Comment by ejfasting** [comment]
Hello @Nawksen,

I'm not sure what you mean, this code works:
```typescript
try
{
    var listAgent = soApi.getListAgent();
    var listOfCountries = [752];
    const result = await listAgent.getCountryListAsync(listOfCountries);
    result.forEach(function (value) {
        context.result.body = value.name;
    }); 
}
catch
{
    context.result.body = "hit the catch";
}
```

So maybe there is something with your custom printLine-function (?)

Can you try my snippet and see if it works for you?

**Reply by PPetterssonSO** [reply]
Sigh, I forgot the await part of the await DoSomeListStuff()

![Paris_Tuileries_Garden_Facepalm_statue](https://github.com/user-attachments/assets/74f3f8d8-7784-4839-a107-84aa0f127c9e)

My bad on that one, thanks for the reply

**Reply by ejfasting** [reply]
Glad i could help! 
No worries, we are taking all comments/Questions as feedback at this point, to try and make it easier to work with.

//Eivind

**Reply by PPetterssonSO** [reply]
Another quick question regarding lists, is it possible to include or exclude deleted listRows?

  var listAgent = soApi.getMDOAgent();
   var response = await listAgent.getSimpleListAsync(listName, <some overload to exclude deleted>);

**Reply by SuperOfficeDevNet** [reply]
Lists exclude deleted rows by default. Include them by adding ShowDeleted to the Additional parameter.

[Forum reference](https://community.superoffice.com/en/technical/forums/api-forums/client-libraries-and-tools/include-deleted-items-from-mdoagent/)

A better alternative is to use GetSelectableList and inspect the Deleted property.

**Reply by ejfasting** [reply]
@Nawksen 

Did you get the answers you were looking for? I'm considering closing this discussion :)

//Eivind
